### DISCUSSION TITLE
Download data by city/country

### DISCUSSION THREAD

**Discussion by etienne-leroux** [OvertureMaps/data#49] [Discussion]
I have downloaded Places data in Parquet format and are trying to figure out which of the Parquet Files contains Places data for a specific city. Are the Parquet files city or country based? I am using DuckDB. Any help will be appreciated. Thank you.

**Comment by songololo** [comment]
You can use `ogr2ogr` to clip the data, for example:
```bash
ogr2ogr -progress -a_srs EPSG:4326 -spat 23.5564 37.7753 24.0717 38.1758 file_name_b.gpkg file_name_a.gpkg
```

At this point the best method to do this directly is probably to filter spatially, for example:
```python
f"""
            COPY (
        SELECT
            id,
            updatetime,
            version,
            JSON(names) as names,
            level,
            height,
            numfloors,
            class,
            -- JSON(sources) as sources,
            -- JSON(bbox) as bbox,
            ST_GeomFromWkb(geometry) AS geometry
        FROM read_parquet('s3://overturemaps-us-west-2/release/2023-07-26-alpha.0/theme=buildings/type=building/*', filename=true, hive_partitioning=1)
        WHERE bbox.minX > {min_x}
            AND bbox.minY > {min_y}
            AND bbox.maxX < {max_x}
            AND bbox.maxY < {max_y}
    ) TO '{out_path.resolve()}/{file_prefix}_buildings.gpkg'
    WITH (FORMAT GDAL, DRIVER 'GPKG')
"""
```
Note that it appears DuckDB still needs to parse through all of the data (i.e. doesn't currently use spatial indices or such)

**Comment by jiayuasu** [comment]
@etienne-leroux Wherobots just released the spatially indexed GeoParquet format of OMF data, done by Apache Sedona. The dataset is free and public on S3, with same schema of the original dataset. With this dataset, a spatial range query on the 120GB Building dataset takes around 3 minute to finish on a single-node 4GB Apache Sedona docker container.

The data is free and public on AWS S3. For more details, please see my blog post on Medium: [medium.com/@dr.jiayu/harnessing-overture-maps-data-apache-sedonas-journey-from-parquet-to-geoparquet-d99f7767a499](https://medium.com/@dr.jiayu/harnessing-overture-maps-data-apache-sedonas-journey-from-parquet-to-geoparquet-d99f7767a499)
